### PR TITLE
Mist 43714 error warning messages should not be shown as config failed

### PR DESCRIPTION
**Pull request by kimberlih**
Added additional tests.

Closes #43714

### CODE REVIEW

**Review by weszeloos — APPROVED** [review]
lgtm

File: netconf/rpc_test.go [hunk 1]
```diff
@@ -156,6 +156,7 @@ func TestUUIDChar(t *testing.T) {
 var RPCReplytests = []struct {
 	rawXML  string
 	replyOk bool
+	Error   bool
```
Comment (charl) [inline netconf/rpc_test.go]: Errors should not be bool, rather use:

```go
	err   error
```

File: netconf/rpc_test.go [hunk 2]
```diff
@@ -216,18 +216,44 @@ configuration check-out failed: (missing mandatory statements)
...
 		false,
 	},
 }
 
 func TestNewRPCReply(t *testing.T) {
 	for _, tc := range RPCReplytests {
 		reply, err := newRPCReply([]byte(tc.rawXML), false, "101")
-		if err != nil {
-			t.Fatalf("unexpected error: %v", err)
+		if (err == nil) == tc.Error {
```
Comment (charl) [inline netconf/rpc_test.go]: This can rather be simplified as:
```go
		if err != tc.Error {
```

**Review by charl — CHANGES_REQUESTED** [review]
Based on https://mistsys.atlassian.net/browse/MIST-43714?focusedCommentId=206579 this issue is not reproducible so I am not convinced we should be changing anything unless we can test it in the wild.

How do we plan to validate this change?

**Comment by kimberlih** [comment]
@charl This change does nothing more than adding more tests to run against a method in go-netconf. There is nothing to test in prod, as long as the tests pass then this "change" it is thoroughly tested.

**Review by weszeloos — APPROVED** [review]
lgtm

File: netconf/rpc_test.go (hunk 1 above)
Comment (kimberlih) [reply netconf/rpc_test.go]: implemented

File: netconf/rpc_test.go (hunk 2 above)
Comment (kimberlih) [reply netconf/rpc_test.go]: modified this to accomodate the new way of comparison.

**Review by charl — APPROVED** [review]
lgtm

**Pull request merged**